toolchain/binutils.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import archive
+from avocado.utils import build
+from avocado.utils import process
+from avocado.utils.software_manager import SoftwareManager
+
+backend = SoftwareManager()
+
+
+class Binutils(Test):
+    """
+    This testcase make use of unit testsuite provided by the
+    source package, performs functional test for all binary tools
+    source file is downloaded and compiled.
+    """
+
+    def setUp(self):
+        '''
+        Install required tools and resolve dependencies
+        '''
+        backend.install('rpmbuild')
+        backend.install('elfutils')
+        backend.install('build')
+        backend.install('autoconf')
+        backend.install('automake')
+        backend.install('binutils-devel')
+        backend.install('dejagnu')
+        backend.install('libtool')
+
+        """
+        Extract - binutils
+        Source: http://ftp.gnu.org/gnu/binutils/binutils-2.25.1.tar.bz2
+        """
+        tarball = self.fetch_asset(
+            'http://ftp.gnu.org/gnu/binutils/binutils-2.25.1.tar.bz2')
+        archive.extract(tarball, self.srcdir)
+
+        bintools_version = os.path.basename(tarball.split('.tar.')[0])
+        self.srcdir = os.path.join(self.srcdir, bintools_version)
+
+    def test(self):
+        os.chdir(self.srcdir)
+
+        process.run('./configure')
+        build.make(self.srcdir)
+        build.make(self.srcdir, extra_args='check')
```

Comment:
This runs the checks and produces huge output, but one does not see the details. So basically this test gives you the PASS/FAIL, but later on you can't tell why it failed. Would you please copy the `*.sum` files from the `binutils` build directory to the test results to allow postprocess?

Additionally it would be convenient to list the failed cases if it fails. Manually I do `for summary in $(find . -name '*.sum'); do echo $summary; grep ^FAIL $summary; done` in the build directory.

So I'd suggest:

```python
for root, dirnames, filenames in os.walk('.'):
    for filename in fnmatch.filter(filenames, '*.sum'):
        filename = os.path.join(root, filename)
        logfile = filename[:-4] + ".log"
        # copy logfile to self.logdir
        # open(filename); read line-by-line and self.log.error lines starting with `FAIL`
        # (alternatively attach the full filename as it might be useful to see what tests were skipped)
```

What do you think?